### ISSUE TITLE
CRASH in burst_threadfilter test

### ISSUE THREAD

**Issue by derekbruening** [DynamoRIO/dynamorio#3335]
Xref #2941.  This crash happened once on CDash:

http://dynamorio.org/CDash/testDetails.php?test=608730&build=40402

```
Running cmd |/work/dr/nightly/run/build_release-external-64/clients/bin64/tool.drcacheoff.burst_threadfilter|
Running postcmd |/work/dr/nightly/run/build_release-external-64/clients/bin64/drcachesim;-indir;/work/dr/nightly/run/build_release-external-64/suite/tests/drmemtrace.tool.drcacheoff.burst_threadfilter.01097.3317.dir;-simulator_type;basic_counts|
CMake Error at /work/dr/nightly/src/suite/tests/runmulti.cmake:106 (message):
  *** postcmd failed (Segmentation fault): Basic counts tool results:

  Total counts:

        546197 total (fetched) instructions
             0 total non-fetched instructions
             0 total prefetches
         54685 total data loads
         17112 total data stores
             4 total threads
            58 total scheduling markers
             0 total transfer markers
             0 total function id markers
             0 total function return address markers
             0 total function argument markers
             0 total function return value markers
             0 total other markers

  Thread 1102 counts:

        182579 (fetched) instructions
             0 non-fetched instructions
             0 prefetches
         18233 data loads
          5693 data stores
            18 scheduling markers
             0 transfer markers
             0 function id markers
             0 function return address markers
             0 function argument markers
             0 function return value markers
             0 other markers

  Thread 1104 counts:

        182053 (fetched) instructions
             0 non-fetched instructions
             0 prefetches
         18228 data loads
          5707 data stores
            18 scheduling markers
             0 transfer markers
             0 function id markers
             0 function return address markers
             0 function argument markers
             0 function return value markers
             0 other markers

  Thread 1100 counts:

        181564 (fetched) instructions
             0 non-fetched instructions
             0 prefetches
         18224 data loads
          5712 data stores
            18 scheduling markers
             0 transfer markers
             0 function id markers
             0 function return address markers
             0 function argument markers
             0 function return value markers
             0 other markers

  Thread 1097 counts:

             1 (fetched) instructions
             0 non-fetched instructions
             0 prefetches
             0 data loads
             0 data stores
             4 scheduling markers
             0 transfer markers
             0 function id markers
             0 function return address markers
             0 function argument markers
             0 function return value markers
             0 other markers

  ***

Call Stack (most recent call first):
  /work/dr/nightly/src/suite/tests/runmulti.cmake:118 (process_cmdline)
 
```

**Comment by hgreving2304** [comment]
I think it is crashing not in DR core but offline in drcachesim postcmd (it's relatively easy to reproduce locally w/ ctest in about 1 every few dozen times.

**Comment by derekbruening** [comment]
Theory: this crash is due to a race which was fixed by adding a missing lock in 26ca7cd9fc9b8a4a704019328695d409722c8e43?

**Comment by hgreving2304** [comment]
Crash is gone starting 26ca7cd9fc9. Closing this one.
